src/clj/fluree/db/flake/reasoner.cljc

```diff
@@ -235,21 +256,47 @@
             (recur r db**))
           db*)))))
 
+(defn find-duplicate-ids
+  [raw-rules]
+  (let [rule-ids (map first raw-rules)]
+    (->> rule-ids
+         frequencies
+         (filter #(< 1 (val %))))))
+
+(defn deduplicate-raw-rules
+  "Given a list of reasoning rules, identifies rules with duplicate ids and renames them using
+  indexes."
+  [raw-rules]
+  (let [duplicate-ids (find-duplicate-ids raw-rules)]
```

Comment:
This binding should have a different name; these are not duplicate ids but a map from duplicate ids to their number of occurrences.